### PR TITLE
Silent warning mentioned on #1307

### PR DESCRIPTION
**Pull request by williampma**

### CODE REVIEW

File: opencog/persist/zmq/events/AtomSpacePublisherModule.cc [hunk 1]
```diff
@@ -365,6 +367,24 @@ Object AtomSpacePublisherModule::tvToJSON(TruthValuePtr tvp)
...
+            jsonDetails.push_back(Pair("confidence", tvp->getConfidence()));
+            json.push_back(Pair("details", jsonDetails));
+            break;
+        }
+
+        case FUZZY_TRUTH_VALUE: {
+            json.push_back(Pair("type", "fuzzy"));
+            jsonDetails.push_back(Pair("strength", tvp->getMean()));
+            jsonDetails.push_back(Pair("count", tvp->getCount()));
+            jsonDetails.push_back(Pair("confidence", tvp->getConfidence()));
```
Comment (misgeatgit) [inline opencog/persist/zmq/events/AtomSpacePublisherModule.cc]: Fuzzy truth value  is sopposed to store only strength and count. https://github.com/opencog/opencog/blob/master/opencog/atomspace/FuzzyTruthValue.h#L41

Comment (williampma) [inline opencog/persist/zmq/events/AtomSpacePublisherModule.cc]: It is suppose to store only strength and conf.

But there's also a count KKK that starts at 800, but the count is actually calculated from the conf.

Anyway, all TV type can output all 3 values, even STV.

Comment (williampma) [inline opencog/persist/zmq/events/AtomSpacePublisherModule.cc]: P.S.  I don't know why STV and FTV's constructor in C++ takes count, when the construction from Scheme takes confidence value.  Even the output is in confidence value.

**Comment by linas** [comment]
I am not sure how many float-point numbers Fuzzy and Probabilistic are "supposed" to have -- either 2 or 3. It currently appear that having 3 might be better (strength, confidence, count) but that would require that some agent or process or function would look at the count every now and then, and update the confidence to match.

By contrast, the current SimpleTV does this "automatically" if you increment the count by 1, then it adds 800, uses that weird formula and comes up with a confidence. I always thought the 800 was arbitrary and hacky, but Ben has an argument that says that 800 is "reasonably OK for most cases."

So: I'm not sure which strategy is preferred: both fuzzy and probabilistic could use two values +800, or they could use 3 values, and some other approach.  You will have to talk to Ben and/or decide what the best solution is.  (I myself don't know)

**Pull request merged**